packages/message-pickup-repository-dynamodb/src/client.ts

```diff
@@ -0,0 +1,208 @@
+import { randomInt } from 'node:crypto'
+import {
+  CreateTableCommand,
+  CreateTableCommandInput,
+  DeleteItemCommand,
+  DeleteItemCommandInput,
+  DescribeTableCommand,
+  DescribeTableCommandInput,
+  DynamoDBClient,
+  DynamoDBClientConfigType,
+  QueryCommand,
+  QueryCommandInput,
+  ScanCommand,
+  ScanCommandInput,
+  UpdateItemCommand,
+} from '@aws-sdk/client-dynamodb'
+import { marshall, unmarshall } from '@aws-sdk/util-dynamodb'
+import { EncryptedMessage } from '@credo-ts/core'
+import { QueuedMessage, attributeDefinitions, keySchema } from './structure'
+
+export type AddQueuedMessageOptions = {
+  connectionId: string
+  receivedAt?: Date
+  recipientDids: string[]
+  encryptedMessage: EncryptedMessage
+}
+
+export type RemoveQueuedMessageOptions = {
+  connectionId: string
+  messageIds: Array<string>
+}
+
+export type DynamodbClientRepositoryOptions = DynamoDBClientConfigType
+
+export class DynamodbClientRepository {
+  private dynamodbClient: DynamoDBClient
+  private tableName = 'queued_messages'
+
+  private constructor(options: DynamoDBClientConfigType) {
+    this.dynamodbClient = new DynamoDBClient(options)
+  }
+
+  public static async initialize(options: DynamoDBClientConfigType): Promise<DynamodbClientRepository> {
+    const dcr = new DynamodbClientRepository(options)
+
+    const params: CreateTableCommandInput = {
+      TableName: dcr.tableName,
+      AttributeDefinitions: attributeDefinitions,
+      KeySchema: keySchema,
+
+      // TODO: correctly define these numbers
+      ProvisionedThroughput: {
+        ReadCapacityUnits: 5,
+        WriteCapacityUnits: 5,
+      },
+    }
+
+    try {
+      const command = new CreateTableCommand(params)
+      await dcr.dynamodbClient.send(command)
+      await dcr.waitForTableToExist()
```

Comment:
Interesting, is this the recommended approach?

Reply:
I think there can be a delay between creating the table and accessing it, this makes sure you can access it directly after creation.